pocs/filterwheel/filterwheel.py

```diff
@@ -176,26 +183,31 @@ def move_to(self, position, blocking=False):
             'g_04'
         """
         assert self.is_connected, self.logger.error("Filter wheel must be connected to move")
+
         if self.camera and self.camera.is_exposing:
             msg = f'Attempt to move filter wheel {self} while camera is exposing, ignoring.'
             raise error.PanError(msg)
 
+        if self.is_moving:
+            msg = f'Attempt to move filter wheel {self} while already moving. Waiting.'
+            self.logger.warning(msg)
+            self._move_event.wait()
+
         position = self._parse_position(position)
         self.logger.info("Moving {} to position {} ({})".format(
             self, position, self.filter_names[position - 1]))
-        move_event = threading.Event()
 
         if position == self.position:
-            # Don't go nowhere
-            move_event.set()
-            return move_event
+            # Already at requested position, don't go nowhere.
+            return self._move_event
 
-        self._move_to(position, move_event)  # Private method to actually perform the move.
+        self._move_event.clear()
+        self._move_to(position)  # Private method to actually perform the move.
 
         if blocking:
-            move_event.wait()
+            self._move_event.wait()
```

Comment:
Ditto on the timeout.

Reply:
There are timeouts in the moves all the derived classes, but they're buried fairly deep down. In the simulator it's in the `_move_to()` method, whereas in the SBIG and ZWO filterwheels they're even more hidden because they're in the movement methods of the driver class. In each case there is a polling thread that includes a timer, and a `finally` clause that should make sure the event always gets set.

Ideally I'd refactor this like I did with the cameras `take_exposure()`, breaking the implementation up into several private methods, with most of it in the base class and only a small amount of hardware specific code in the derived classes. A job for another PR though, I think.